### PR TITLE
updated for Mailbox API 2.0

### PR DESCRIPTION
**Pull request by gurs1kh**

### CODE REVIEW

File: lib/help_scout.rb [hunk 1]
```diff
@@ -229,23 +230,37 @@ def search(path, query, page_id = 1, items = [])
...
+    token = JSON.parse(response.body)['access_token']
+    "Bearer #{token}"
+  end
+
   def request(method, path, options)
-    uri = URI("https://api.helpscout.net/v1/#{path}.json")
+    uri = URI("https://api.helpscout.net/v2/#{path}")
 
     # The password can be anything, it's not used, see:
     # http://developer.helpscout.net/help-desk-api/
```
Comment (NCSkoglund) [inline lib/help_scout.rb]: @gurs1kh might want to also remove this comment ✂️

Comment (gurs1kh) [reply lib/help_scout.rb]: Thanks! I've fixed it in the latest commit

**Comment by bittersweet** [comment]
Hey @gurs1kh and @NCSkoglund , thanks for contributing! We were planning on starting work on the update this week or the next, and it looks like we'll be able to merge this to start with a good base, so thanks a lot :-) 

One thing I/we quickly noticed was the authorization part. From looking over [the docs](https://developer.helpscout.com/mailbox-api/overview/authentication/), the token is valid for 2 hours:

> This token is valid for 2 hours and you should create a new one only after the existing token expires. Expiration will be indicated by the API responding with HTTP 401.

They respond with a `401` if the token is expired, and HS has a way to refresh your token as well. That is something that will be interesting to implement instead of getting a new token every time.

**Comment by bittersweet** [comment]
Hey @gurs1kh , I've merged your changes into a new temporary branch named `api_v2`. I'll be doing more work on this branch starting this week. I'll try to keep everyone updated in #17 !

**Pull request merged**